packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -977,7 +987,7 @@ function workLoop(isYieldy) {
     }
   } else {
     // Flush asynchronous work until the deadline runs out of time.
-    while (nextUnitOfWork !== null && !shouldYield()) {
+    while (nextUnitOfWork !== null && (!shouldYield() || nextRenderDidExpire)) {
```

Comment:
TODO: Meh, it's probably fine to render non-async, strict mode placeholders asynchronously. Revert.